include/vrv/scoringupfunctor.h

```diff
@@ -0,0 +1,118 @@
+
+    /*
+     * Methods
+     */
+    /**
+     * @name: Divide the notes of a voice into sequences to be processed individualy
+     */
+    ///@{
+    void workInMensur(const ArrayOfElementDurPairs &m_dursInVoiceSameMensur, data_DURATION noteLevel);
+    std::vector<ArrayOfElementDurPairs> SubdivideIntoBoundedSequences(
+        const ArrayOfElementDurPairs &dursInVoiceSameMensur, data_DURATION boundUnit);
+    void ProcessBoundedSequences(const std::vector<ArrayOfElementDurPairs> &listOfSequences, data_DURATION boundUnit);
+    void ProcessBoundedSequences(const ArrayOfElementDurPairs &sequence, data_DURATION boundUnit);
+    ArrayOfElementDurPairs GetBoundedNotes(const ArrayOfElementDurPairs &sequence, data_DURATION boundUnit);
+    ///@}
+
+    /**
+     * @name: Apply the principles of imperfection and alteration (find @dur.quality of notes: perfecta / imperfecta /
+     * altera) for sequences with or without dots of division
+     */
+    ///@{
+    bool EvalDotOfDiv(const ArrayOfElementDurPairs &middleSeq, const ArrayOfElementDurPairs &sequence, int dotInd,
+        data_DURATION unit);
+    void FindDurQuals(const ArrayOfElementDurPairs &middleSeq, double valueInUnit, data_DURATION boundUnit);
+    ///@}
+
+    /**
+     * @name: Find the duration value of the note in minims or another given unit
+     */
+    ///@{
+    double GetDurNumberValue(
+        const std::pair<LayerElement *, data_DURATION> &elementDurPair, bool followedByDot, LayerElement *nextElement);
+    double GetValueInMinims(const ArrayOfElementDurPairs &middleSeq);
+    double GetValueInUnit(double valueInMinims, data_DURATION unit);
+    ///@}
+
+    /**
+     * @name Apply the modifications of imperfection and alteration or leaves the notes with their default perfect value
+     */
+    ///@{
+    Note *ImperfectionAPP(const ArrayOfElementDurPairs &sequence, data_DURATION boundUnit);
+    Note *ImperfectionAPA(const ArrayOfElementDurPairs &sequence, data_DURATION boundUnit);
+    Note *Alteration(const ArrayOfElementDurPairs &sequence, data_DURATION boundUnit);
+    bool LeavePerfect(const ArrayOfElementDurPairs &sequence, data_DURATION boundUnit);
+    void ApplyAugmentationsAndPerfections();
+    ///@}
+
+    /*
+     * Functor interface
+     */
+    ///@{
+    FunctorCode VisitLayer(Layer *layer) override;
+    FunctorCode VisitLayerElement(LayerElement *layerElement) override;
+    ///@}
+    ///
+
+protected:
+    //
+private:
+    int m_modusMaior;
+    int m_modusMinor;
+    int m_tempus;
+    int m_prolatio;
+    ArrayOfElementDurPairs m_dursInVoiceSameMensur;
+    std::vector<ArrayOfElementDurPairs> m_listOfSequences;
+    std::list<std::pair<Note *, Dot *>> m_listOfAugNotesDotsPairs;
+    std::list<std::pair<Note *, Dot *>> m_listOfPerfNotesDotsPairs;
+
+public:
+    //
+
+private:
+    // The current score time in the measure (incremented by each element)
+    double m_currentScoreTime;
```

Comment:
Not sure where you use this but you might want to change to `Fraction` introduced [here](https://github.com/rism-digital/verovio/pull/3806).

Reply:
True, I am going to use this for adding the barlines automatically. Once the implementation of the scoring up is complete and correct, I will do the automatic barring stuff.